# Wanjiku's Gikomba Stall Sales Analysis

## Task 1 — Sales Performance

I calculate the total, average, highest, and lowest sales for the 14-day period. I also identify the day name and week number for the highest and lowest sales.

In [1]:
daily_sales = [2300, 1950, 2800, 2100, 3600, 5200, 4100,
               2500, 2200, 3100, 2400, 3900, 5600, 4700]   # 14 days, KES

days = ["Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"] * 2

# qty = kg in stock, reorder = reorder level (kg), cost/price = KES per kg
stock = {
    "Tomatoes": {"qty": 18, "reorder": 25, "cost": 60, "price": 90},
    "Onions":   {"qty": 40, "reorder": 20, "cost": 70, "price": 100},
    "Sukuma":   {"qty": 9,  "reorder": 15, "cost": 30, "price": 50},
    "Potatoes": {"qty": 55, "reorder": 30, "cost": 50, "price": 75},
    "Avocado":  {"qty": 12, "reorder": 10, "cost": 80, "price": 140},
}

In [2]:
total_sales = sum(daily_sales)
average_sales = total_sales / len(daily_sales)

highest_sales = max(daily_sales)
lowest_sales = min(daily_sales)

highest_index = daily_sales.index(highest_sales)
lowest_index = daily_sales.index(lowest_sales)

highest_day = days[highest_index]
lowest_day = days[lowest_index]

highest_week = highest_index // 7 + 1
lowest_week = lowest_index // 7 + 1

print(f"Total sales: KES {total_sales:,}")
print(f"Average daily sales: KES {average_sales:,.2f}")
print(f"Highest sales: KES {highest_sales:,} on {highest_day}, Week {highest_week}")
print(f"Lowest sales: KES {lowest_sales:,} on {lowest_day}, Week {lowest_week}")

Total sales: KES 46,450
Average daily sales: KES 3,317.86
Highest sales: KES 5,600 on Sat, Week 2
Lowest sales: KES 1,950 on Tue, Week 1


## Task 2 — Weekly Sales Growth

I create a week_totals() function that uses slicing to calculate the total sales for Week 1 and Week 2. I then calculate the percentage growth from Week 1 to Week 2.

In [3]:
def week_totals(sales):
    """Return the total sales for Week 1 and Week 2."""
    week1 = sum(sales[:7])
    week2 = sum(sales[7:])
    return week1, week2


week1_total, week2_total = week_totals(daily_sales)

growth_percentage = ((week2_total - week1_total) / week1_total) * 100

print(f"Week 1 total: KES {week1_total:,}")
print(f"Week 2 total: KES {week2_total:,}")
print(f"Week-on-week growth: {growth_percentage:.2f}%")

Week 1 total: KES 22,050
Week 2 total: KES 24,400
Week-on-week growth: 10.66%


## Task 3 — Daily Sales Classification

I classify each day as Strong, Normal, or Weak based on its sales compared with the overall average. I then count how many days belong to each category using a dictionary.

In [4]:
def classify_day(amount, average):
    """Classify a sales amount as Strong, Normal, or Weak compared with the average."""
    if amount > average * 1.20:
        return "Strong"
    elif amount < average * 0.80:
        return "Weak"
    else:
        return "Normal"


category_counts = {
    "Strong": 0,
    "Normal": 0,
    "Weak": 0
}

daily_categories = []

for day, amount in zip(days, daily_sales):
    category = classify_day(amount, average_sales)
    daily_categories.append(category)
    category_counts[category] += 1
    print(f"{day}: KES {amount:,} - {category}")

print("\nCategory counts:")
print(category_counts)

Mon: KES 2,300 - Weak
Tue: KES 1,950 - Weak
Wed: KES 2,800 - Normal
Thu: KES 2,100 - Weak
Fri: KES 3,600 - Normal
Sat: KES 5,200 - Strong
Sun: KES 4,100 - Strong
Mon: KES 2,500 - Weak
Tue: KES 2,200 - Weak
Wed: KES 3,100 - Normal
Thu: KES 2,400 - Weak
Fri: KES 3,900 - Normal
Sat: KES 5,600 - Strong
Sun: KES 4,700 - Strong

Category counts:
{'Strong': 4, 'Normal': 4, 'Weak': 6}


## Task 4 — Weekday versus Weekend Sales

I separate the sales into weekdays (Monday to Friday) and weekends (Saturday and Sunday). I calculate the average for each group and the percentage difference between weekend and weekday averages.

In [5]:
weekday_sales = []
weekend_sales = []

for day, amount in zip(days, daily_sales):
    if day in ["Mon", "Tue", "Wed", "Thu", "Fri"]:
        weekday_sales.append(amount)
    else:
        weekend_sales.append(amount)

weekday_average = sum(weekday_sales) / len(weekday_sales)
weekend_average = sum(weekend_sales) / len(weekend_sales)

weekend_difference = ((weekend_average - weekday_average) / weekday_average) * 100

print(f"Weekday average: KES {weekday_average:,.2f}")
print(f"Weekend average: KES {weekend_average:,.2f}")
print(f"Weekend sales are {weekend_difference:.2f}% higher than weekday sales.")

Weekday average: KES 2,685.00
Weekend average: KES 4,900.00
Weekend sales are 82.50% higher than weekday sales.


## Task 5 — Stock Reordering and Gross Margins

I create a function that checks the stock dictionary and returns products whose quantity is at or below their reorder level. I also calculate the gross margin percentage for every product using (price - cost) / price.

In [6]:
def products_to_reorder(stock_data):
    """Return products whose quantity is at or below their reorder level."""
    reorder_list = []

    for product, details in stock_data.items():
        if details["qty"] <= details["reorder"]:
            reorder_list.append(product)

    return reorder_list


reorder_products = products_to_reorder(stock)

print("Products to reorder:")
for product in reorder_products:
    print(f"- {product}")

print("\nGross margins:")

for product, details in stock.items():
    margin = ((details["price"] - details["cost"]) / details["price"]) * 100
    print(f"{product}: {margin:.2f}%")

Products to reorder:
- Tomatoes
- Sukuma

Gross margins:
Tomatoes: 33.33%
Onions: 30.00%
Sukuma: 40.00%
Potatoes: 33.33%
Avocado: 42.86%


## Task 6 — Stall Performance Report

I combine the main findings into a formatted business report using Python f-strings. Sales and monetary values are displayed in Kenyan Shillings with thousands separators.

In [7]:
print("=" * 50)
print("          STALL PERFORMANCE REPORT")
print("=" * 50)

print(f"Total sales:          KES {total_sales:,}")
print(f"Average daily sales:  KES {average_sales:,.2f}")
print(f"Week 1 sales:         KES {week1_total:,}")
print(f"Week 2 sales:         KES {week2_total:,}")
print(f"Sales growth:         {growth_percentage:.2f}%")
print(f"Best sales day:       {highest_day}, Week {highest_week}")
print(f"Best sales amount:    KES {highest_sales:,}")
print(f"Worst sales day:      {lowest_day}, Week {lowest_week}")
print(f"Worst sales amount:   KES {lowest_sales:,}")
print(f"Weekday average:      KES {weekday_average:,.2f}")
print(f"Weekend average:      KES {weekend_average:,.2f}")
print(f"Weekend advantage:    {weekend_difference:.2f}%")
print(f"Days classified Strong: {category_counts['Strong']}")
print(f"Days classified Normal: {category_counts['Normal']}")
print(f"Days classified Weak:   {category_counts['Weak']}")
print(f"Products to reorder:  {', '.join(reorder_products)}")

print("=" * 50)

          STALL PERFORMANCE REPORT
Total sales:          KES 46,450
Average daily sales:  KES 3,317.86
Week 1 sales:         KES 22,050
Week 2 sales:         KES 24,400
Sales growth:         10.66%
Best sales day:       Sat, Week 2
Best sales amount:    KES 5,600
Worst sales day:      Tue, Week 1
Worst sales amount:   KES 1,950
Weekday average:      KES 2,685.00
Weekend average:      KES 4,900.00
Weekend advantage:    82.50%
Days classified Strong: 4
Days classified Normal: 4
Days classified Weak:   6
Products to reorder:  Tomatoes, Sukuma


## Task 7 — Business Recommendations

The recommendations below are based directly on the sales, growth, weekend performance, stock, and margin results.

# Recommendations for Wanjiku

Wanjiku should prioritize weekend sales because the average weekend sales of KES 4,900 are 82.50% higher than the weekday average of KES 2,685. The business should restock Tomatoes and Sukuma because their current quantities of 18 kg and 9 kg are below their reorder levels of 25 kg and 15 kg respectively. Sales grew by 10.66% from KES 22,050 in Week 1 to KES 24,400 in Week 2, so Wanjiku should continue monitoring sales to maintain this growth. Wanjiku should also pay attention to Avocado because it has the highest gross margin of 42.86%, which is higher than the margins of the other products.